source/Sashimi.Aws.Tests/CloudFormation/CloudFormationActionHandlerFixture.cs

```diff
@@ -0,0 +1,319 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using FluentAssertions;
+using NuGet.Packaging;
+using NuGet.Versioning;
+using NUnit.Framework;
+using Sashimi.Aws.ActionHandler;
+using Sashimi.Tests.Shared;
+using Sashimi.Tests.Shared.Server;
+
+namespace Sashimi.Aws.Tests.CloudFormation
+{
+    public class CloudFormationActionHandlerFixture
+    {
+        const string AwsStackRole = "arn:aws:iam::968802670493:role/e2e_buckets";
+        // BucketName must use the following prefix, otherwise the above stack role will not have permission to access 
+        const string ValidBucketNamePrefix = "cfe2e-";
+        const string TransformIncludeLocation = "s3://octopus-e2e-tests/permanent/tags.json";
+        const string AwsRegion = "us-east-1";
+
+        string stackName;
+
+        [SetUp]
+        public void Setup()
+        {
+            stackName = $"E2ETestStack-{UniqueName.Generate()}";
+        }
+
+        [Test]
+        public void RunCloudFormation_InlineSourceWithoutParameters()
+        {
+            var bucketName = $"{ValidBucketNamePrefix}{UniqueName.Short()}";
+
+            var template = File.ReadAllText(Path.Combine(TestEnvironment.GetTestPath(), "CloudFormation", "package-withoutparameters", "template.json"))
+                .Replace("#{BucketName}", bucketName);
+            var result = ActionHandlerTestBuilder.Create<AwsRunCloudFormationActionHandler, Calamari.Aws.Program>()
+                    .WithArrange(context =>
+                    {
+                        context.WithStack(stackName)
+                            .WithAwsAccount()
+                            .WithAwsRegion(AwsRegion)
+                            .WithStackRole(AwsStackRole)
+                            .WithAwsTemplateInlineSource(template, null);
+                        
+                        context.Variables.Add(AwsSpecialVariables.Action.Aws.WaitForCompletion, bool.TrueString);
+                    })
+                    .Execute(false);
+
+            result.WasSuccessful.Should().BeTrue();
+            result.OutputVariables["AwsOutputs[OutputName]"].Value.Should().Be(bucketName);
+        }
+
+        [Test]
+        public void RunCloudFormation_InlineSourceWithYamlTemplateAndParameters()
+        {
+            var nameVarParamValue = $"{ValidBucketNamePrefix}{UniqueName.Short()}";
+            var namePlainParamValue = $"{ValidBucketNamePrefix}{UniqueName.Short()}";
+
+            var templateFolderPath = Path.Combine(TestEnvironment.GetTestPath(), "CloudFormation", "package-withparameters");
+            var template = File.ReadAllText(Path.Combine(templateFolderPath, "template.yaml"));
+            var parameters = File.ReadAllText(Path.Combine(templateFolderPath, "parameters.json"))
+                .Replace("#{NameVarParamValue}", nameVarParamValue)
```

Comment:
The variable `NameVarParam` should not be replaced in the parameters. We are relying on Octopus to do the parameter substitution. Same deal with `RunCloudFormation_PackageWithParameters()`.

Reply:
Are we referring to the actual `Octopus Server` here? If so, we don't have `Octopus Server` setup for these e2e tests.

Reply:
I suspect it's inside the command: https://github.com/OctopusDeploy/Sashimi/blob/345049adb08aacfc5d96dfba4c59d155fac01d74/source/Calamari.Aws/DeployAwsCloudFormationCommand.cs#L104

Commenting out L137 makes the test fail, then adding L155 makes it pass again
![image](https://user-images.githubusercontent.com/2888279/82847524-eb699300-9f31-11ea-95c1-c9383a5464ec.png)